docs/reference/resources/Microsoft/Windows/RebootPending/index.md

```diff
@@ -0,0 +1,139 @@
+---
+description: Microsoft.Windows/RebootPending resource reference documentation
+ms.date:     03/25/2025
+ms.topic:    reference
+title:       Microsoft.Windows/RebootPending
+---
+
+# Microsoft.Windows/RebootPending
+
+## Synopsis
+
+Checks if a Windows system has a pending reboot.
+
+> [!IMPORTANT]
+> The `Microsoft.Windows/RebootPending` resource are a proof-of-concept example
+> for use with DSC. Don't use it in production.
+
+## Metadata
+
+```yaml
+Version    : 0.1.0
+Kind       : resource
+Tags       : [Windows]
+Author     : Microsoft
+```
+
+## Instance definition syntax
+
+```yaml
+resources:
+  - name: <instance name>
+    type: Microsoft.Windows/RebootPending
+    properties: {}
+```
+
+## Description
+
+The `Microsoft.Windows/RebootPending` resource enables you to check whether a Windows system has a pending reboot. The resource can determine if a system reboot is required due to:
+
+- Windows Updates
+- Component-Based Servicing
+- Pending file rename operations
+- Pending computer rename
+- Pending domain join operations
```

Comment:
I wonder whether it's worth indicating that the resource doesn't tell you _which_ of these reasons is causing a pending reboot, only that one of them is.